src/nemos/glm.py

```diff
@@ -1158,6 +1159,112 @@ def __sklearn_clone__(self) -> GLM:
         klass = self.__class__(**params)
         return klass
 
+    def save_params(self, filename: Union[str, Path]):
+        """
+        Save GLM model parameters to a .npz file.
+
+        This method allows to reuse the model parameters. The saved parameters can be loaded back
+        into a GLM instance using the `load_params` function.
+
+        Parameters
+        ----------
+        filename :
+            The name of the file where the model parameters will be saved. The file will be saved in `.npz` format.
+
+        Examples
+        --------
+        >>> import nemos as nmo
+
+        >>> # Create a GLM model with specified parameters
+        >>> solver_args = {"stepsize": 0.1, "maxiter": 1000, "tol": 1e-6}
+        >>> model = nmo.glm.GLM(
+        ...     regularizer="Ridge",
+        ...     regularizer_strength=0.1,
+        ...     observation_model="Gamma",
+        ...     solver_name="BFGS",
+        ...     solver_kwargs=solver_args,
+        ... )
+
+        >>> # Print the model parameters
+        >>> print(model.get_params())
+
+        >>> # Save the model parameters to a file
+        >>> model.save_params("model_params.npz")
+
+        >>> # Load the model from the saved file
+        >>> model = nmo.load_model("model_params.npz")
+
+        >>> # Print the parameters of the loaded model
+        >>> # This should match the original model parameters
+        >>> print(model.get_params())
+        """
+
+        # initialize saving dictionary
+        fit_attrs = {}
+
+        fit_attrs["coef_"] = self.coef_
+        fit_attrs["intercept_"] = self.intercept_
+
+        string_attrs = ["inverse_link_function"]
+
+        super().save_params(filename, fit_attrs, string_attrs)
+
+    @classmethod
+    def _load_from_dict(
```

Comment:
I think this is a nice solution, but we should document somewhere (for ourselves), that this is how `load_model` works, so we know that we need to write this class method for every new model we write